File: src/GlassCS/Glass/Actions/ToggleExpandCollapseAction.cs
using System.Windows;
using System.Windows.Interactivity;
using ESRI.ArcGIS.SilverlightMapApp;

namespace ESRI.ArcGIS.SilverlightMapApp.Actions
{
    public class ToggleExpandCollapseAction : TargetedTriggerAction<UIElement>
    {
        protected override void Invoke(object parameter)
        {
            GlassPanel panel = this.Target as GlassPanel;
            if (panel.IsOpen)
                panel.Collapse();
            else
                panel.Expand();
        }
    }
}

File: src/GlassVB/GlassVB/Actions/ToggleExpandCollapseAction.vb
Imports System.Windows
Imports System.Windows.Interactivity
Imports ESRI.ArcGIS.SilverlightMapApp

Namespace ESRI.ArcGIS.SilverlightMapApp.Actions
		Public Class ToggleExpandCollapseAction
			Inherits TargetedTriggerAction(Of UIElement)
				Protected Overrides Sub Invoke(ByVal parameter As Object)
						Dim panel As GlassPanel = TryCast(Me.Target, GlassPanel)
						If panel.IsOpen Then
								panel.Collapse()
						Else
								panel.Expand()
						End If
				End Sub
		End Class
End Namespace
